### ISSUE TITLE
Installer Remembers Removed Mods and Lacks Uninstall operations

### ISSUE THREAD

**Issue by mikeloeven** [scp-fs2open/fsoinstaller#1]
So reinstalling FSO when i selected standard i noticed that alot of mods i had previously played were checked and greyed out as if they had already been installed. I was reinstalling from scratch to a new directory so somewhere along the line the installer stored a list of previously installed mods and doesent seem to realize that the old installation was deleted and no longer present have no idea how to reset this or delete the existing application data for the installer 

The installer should also function as a mod manager with the ability to remove installed mods as well as add them

**Comment by asarium** [comment]
Have you tried continuing the installation? The installer should detect that the files of the mods are not present and download them again.
IIRC @Goober5000 mentioned that the mechanism for detecting already installed mods will be improved in the future but currently it just saves which mods have already been installed.

**Comment by MageKing17** [comment]
> Have you tried continuing the installation? The installer should detect that the files of the mods are not present and download them again.

Not unless there's a newer version of the mod available or you check the "Re-run installation for mods that are already installed and up to date" checkbox; the first thing it checks is if the version listed online is the same as the version listed in fsoinstaller.properties and, if they match, it makes no further effort.

> have no idea how to reset this or delete the existing application data for the installer

C:\Users\your_name_here\fsoinstaller.properties

**Comment by mikeloeven** [comment]
the issue i had was after uninstalling freespace2 the mods were still left in the folder so i deleted the fs2 folder and reinstalled. i wanted to actually deselect some of the mods i wasn't planning on using but the installer greyed them all out and i could not deselect them when reinstalling 

i assume deleting fsoinstaller.properties will reset it to first run conditions ?

IT seems like from a programming point of view the installer needs to check if the files for installed mods still exist on the target directory and mark the mod as uninstalled if they are missing or incomplete

**Comment by niffiwan** [comment]
Yes, deleting fsoinstaller.properties will reset it to first run conditions.

**Comment by mikeloeven** [comment]
In the future C:\Users\your_name_here is not really a good place to put program data. best practice is to place these types of file in appdata/roaming under a folder related to the program. In future versions, the file should be located at %appdata%\FSOInstaller\fsoinstaller.properties

If this was the case I could have easily found this file without having to ask for it's location

**Comment by Goober5000** [comment]
The above posters have pretty well covered the essential parts of the questions.  In regards to the location, the installer is cross-platform.  All systems have a home directory, but only Windows systems have %APPDATA%.

As to the primary subject of this ticket -- why the Installer operates this way -- this is because it intentionally copies the design of the original FSO Installer which was written by Turey and used from 2006 to 2010.  Turey's files were instruction-based (do this sequence of operations) rather than asset-based (make sure this file is there) which has the advantage that it's easy to understand but the disadvantage that it's hard to reverse.

(Turey's installer stored the list of installed mods in the FreeSpace2 folder itself, which probably explains why this issue never came up during that time -- users who deleted the folder would have the mod install status reset.)

Now, I do have some ideas on how the Installer can detect mods that are uninstalled.  But this is a low priority at the moment because there are a large number of more pressing matters that require my attention these days, and simply deleting the .properties file is an easy solution.
